terraso_backend/apps/shared_data/visualization_tileset_tasks.py

```diff
@@ -0,0 +1,161 @@
+import csv
+import json
+import threading
+
+import pandas
+import structlog
+from django.conf import settings
+
+from apps.core.gis.mapbox import create_tileset, remove_tileset
+from apps.shared_data.services import data_entry_upload_service
+
+from .models import VisualizationConfig
+
+logger = structlog.get_logger(__name__)
+
+
+class AsyncTaskHandler:
+    def start_task(self, method, args):
+        t = threading.Thread(target=method, args=[*args], daemon=True)
+        t.start()
+
+
+def start_create_mapbox_tileset_task(visualization_id):
+    AsyncTaskHandler().start_task(create_mapbox_tileset, [visualization_id])
+
+
+def start_remove_mapbox_tileset_task(tileset_id):
+    AsyncTaskHandler().start_task(remove_mapbox_tileset, [tileset_id])
+
+
+def get_rows_from_file(data_entry):
+    type = data_entry.resource_type
+    if type.startswith("csv"):
+        file = data_entry_upload_service.get_file(data_entry.s3_object_name, "rt")
+        reader = csv.reader(file)
+        return [row for row in reader]
+    elif type.startswith("xls"):
+        file = data_entry_upload_service.get_file(data_entry.s3_object_name, "rb")
```

Comment:
```suggestion
    file = data_entry_upload_service.get_file(data_entry.s3_object_name, "rb")
    if type.startswith("csv"):
        reader = csv.reader(file)
        return [row for row in reader]
    elif type.startswith("xls"):
```

Reply:
@paulschreiber that refactor is not possible because the read mode is different, one read the file as text `rt` and the other one as binary `rb`

Reply:
I know you can't read binary files in mode, but I believe you can read text files in binary mode, no?

Reply:
Let me try that, not sure

Reply:
@paulschreiber No, csv.reader() doesn't accept a binary file

Reply:
From the [documentation](https://docs.python.org/3/library/csv.html#csv.reader):

> csvfile can be any object which supports the [iterator](https://docs.python.org/3/glossary.html#term-iterator) protocol and returns a string each time its __next__() method is called